Sprint 10

byBlinque opened the pull request.

byBlinque committed:
Сверстал лэйаут трэк итема, добавил стили

byBlinque committed:
Добавил обрезку текста при большой длине

byBlinque committed:
Добавил RV, раздал всем id

byBlinque committed:
Реализовал search track viewholder и адаптер

byBlinque committed:
Подключил библиотеку Glide, доработал recycler view

ilshat-abdulin commented on app/src/main/java/com/example/playlistmakernew/Track.kt:
```diff
@@ -0,0 +1,10 @@
+    var artistName: String, // Имя исполнителя
+    var trackTime: String, // Продолжительность трека
+    var artworkUrl100: String // Ссылка на изображение обложки
+) {
```
🍏 Классы, которые представляют из себя модели данных, лучше объявлять как data class. В data class автоматически переопределяются (используя свойства объявленные в конструкторе) методы hashCode(), equals(), toString()

ilshat-abdulin commented on app/src/main/java/com/example/playlistmakernew/SearchActivity.kt:
```diff
@@ -27,6 +29,34 @@ class SearchActivity : AppCompatActivity() {
+                "https://is5-ssl.mzstatic.com/image/thumb/Music125/v4/a0/4d/c4/a04dc484-03cc-02aa-fa82-5334fcb4bc16/18UMGIM24878.rgb.jpg/100x100bb.jpg")
+        )
+
+        val searchTrackRV: RecyclerView = findViewById(R.id.search_ac_rv)
```
🍏 RecyclerView можно объявить как глобальную переменную также, как и остальные ссылки на view-компоненты, для поддержания единообразия

ilshat-abdulin commented on app/src/main/java/com/example/playlistmakernew/SearchTrackViewHolder.kt:
```diff
@@ -0,0 +1,29 @@
+        trackTime.text = track.trackTime
+        Glide.with(itemView.context)
+            .load(track.artworkUrl100)
+            .placeholder(R.drawable.ic_launcher_background)
```
⚠ Может не обратил внимание: используй placeholder согласно идее в дизайн-проекте
![Снимок экрана 2023-03-02 182522](https://user-images.githubusercontent.com/72722086/222441492-9edb682b-b29b-46b6-a5e9-682211079fa0.png)

ilshat-abdulin replied:
Чтобы не вводить в заблуждение: на платформе я отметил, как критическое замечание, не тот пункт. На самом деле это замечание относится пункту о соответствии дизайн-проекту.

![Снимок экрана 2023-03-02 183807](https://user-images.githubusercontent.com/72722086/222444380-2016cedc-5e3c-46e5-a576-a36a8ac5e7e5.png)

ilshat-abdulin commented on app/src/main/res/layout/search_track_item.xml:
```diff
@@ -0,0 +1,76 @@
+                android:layout_height="wrap_content"
+                android:ellipsize="end"
+                android:lines="1"
+                tools:text="Smells Like Teen Spirit" />
```
Использование tools хорошая практика для проверки макета 👍

byBlinque committed:
Сделал класс Track дата классом
Объявил RV рядом с остальными переменными в начале
Нашел этот незаметный плейсхолдер в макете и заменил на него)
